### PR TITLE
fixed shortcut issue

### LINKED ISSUE #3957: GET_CONTENT shortcut breaks the BottomBar
**Describe the bug**

Navigating to Downloads / OnlineLibrary fragment through **Get Content** shortcut breaks the BottomNavBar. Clicking on the Reader tab doesn't take us to the Reader fragment.

**Expected behavior**

BottoNavigationBar shouldn't break

**Steps to reproduce the behavior:**
1. click on Get Content shortcut
2. click on Reader tab.
3. User cannot open the Reader fragment :(

**Screenshots**

https://github.com/user-attachments/assets/1dea844b-52a6-492d-b3c6-157715ecddc0

**Environment**
- Version of Kiwix Android : 3.11.1
- Device : Real me Narzo 60x
- OS version : 13

### PR DESCRIPTION
**Pull request by shashank-0-0**
Fixes #3957 

The issue was with navigateTo(), which breaks the back stack (https://stackoverflow.com/questions/71525493/bottom-navigation-bar-malfunctions-when-navigating-from-a-fragment)
so used the NavigationUI API to navigate.

**Screenshots** 

https://github.com/user-attachments/assets/205d297f-6e4f-4217-9010-cb733e52dec3

ready to rework if changes are needed.

### CODE REVIEW

**Comment by kelson42** [comment]
@MohitMaliDeveloper What is the status here?

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 1]
```diff
@@ -104,7 +105,9 @@ class KiwixMainActivity : CoreMainActivity() {
     activityKiwixMainBinding = ActivityKiwixMainBinding.inflate(layoutInflater)
     setContentView(activityKiwixMainBinding.root)
     if (intent.action == "GET_CONTENT") {
-      navigate(R.id.downloadsFragment)
```
Comment (MohitMaliFtechiz) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:107]: @shashank-0-0 The back stack is not broken, if I click on the back button it opens the `reader` screen from the back stack so the issue is something else. We should fix the actual issue.


https://github.com/user-attachments/assets/c8f5d277-8090-4126-8fed-200ca0738dd7

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@shashank-0-0 Thank you for the PR, can you please also fix the `GET_CONTENT` shortcut not working when the application is in the background? It works when the application is freshly opened, but not if I open it when the application is in background.


You need handle this intent in the `onNewIntent` method in `KiwixMainActivity. Also can you please write the automated test case for this fix so that we can avoid this type of errors in future. If you need any kind of help in testing let me know.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt (hunk 1 above)
Comment (shashank-shetty-0-0) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:107]: Looking in to it !!

Comment (shashank-0-0) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:107]: looking in to it !!

Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:107]: @shashank-0-0 Any update here?

Comment (shashank-0-0) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:107]: @MohitMaliFtechiz my bad couldn't take the time ,
 i think this is precisely the issue here https://stackoverflow.com/a/71566840/15331061 . Our bottom nav items each have their own backstacks and when we navigateTo(a fragment) we just add the fragment to the respective back stack.
In our case we are  adding the download fragment to the Reader fragment's back stack so when we click on Reader tab we are taken to the top of its backstack which has Download fragment.
So i guess the fix is fine , can u look into it again thanks.

Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:121]: @shashank-0-0 Thank you for your explanation i will have a look at this, in the meantime, can you please implement my other comment?

**Comment by shashank-0-0** [comment]
> @shashank-0-0 Thank you for the PR, can you please also fix the `GET_CONTENT` shortcut not working when the application is in the background? It works when the application is freshly opened, but not if I open it when the application is in background.
> 
> You need handle this intent in the `onNewIntent` method in `KiwixMainActivity`. Also can you please write the automated test case for this fix so that we can avoid this type of errors in future. If you need any kind of help in testing let me know.

@MohitMaliFtechiz please correct me if i am wrong , Isn't oncreate always called when you open a static shortcut ? so is it necessary to handle it in onNewIntent ?

https://github.com/user-attachments/assets/63b66b9b-1edd-4054-b574-9e302c326f0e

**Comment by MohitMaliFtechiz** [comment]
@shashank-0-0 No, `onCreate` will not always call when the application is already in the background. It will only be called when the application is fully closed(not in the background). When it is in the background and we open a static shortcut,t the `onNewIntent` method will call.




https://github.com/user-attachments/assets/a10a3782-4e82-4f64-a1f5-edec483e9b48

**Comment by shashank-0-0** [comment]
> @shashank-0-0 No, `onCreate` will not always call when the application is already in the background. It will only be called when the application is fully closed(not in the background). When it is in the background and we open a static shortcut,t the `onNewIntent` method will call.
> 
>  media_20241104_155720_4821255378590183623.mp4

my bad , Got it pushing the change.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 2]
```diff
@@ -117,9 +118,7 @@ class KiwixMainActivity : CoreMainActivity() {
     super.onCreate(savedInstanceState)
     activityKiwixMainBinding = ActivityKiwixMainBinding.inflate(layoutInflater)
     setContentView(activityKiwixMainBinding.root)
-    if (intent.action == "GET_CONTENT") {
-      navigate(R.id.downloadsFragment)
-    }
+    handleShortcutIntent(intent)
```
Comment (MohitMaliFtechiz) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:121]: The name of the function should be more clear. Currently, the name sounding it is for handling all shortcuts but it is only handling the `GET_CONTENT` shortcut so the name should be `handleGetContentIntent`.

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
Thanks for the changes. Can you please write the automated test case for this? I will be available for any help if you need in this.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt (hunk 2 above)
Comment (shashank-0-0) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt]: my bad ,done !!

**Comment by shashank-0-0** [comment]
> Thanks for the changes. Can you please write the automated test case for this? I will be available for any help if you need in this.

okay!!

**Comment by shashank-0-0** [comment]
@MohitMaliFtechiz I have written instrumented test for getContent shortcut can u review it, and let me know if changes are needed .

**Comment by MohitMaliFtechiz** [comment]
@shashank-0-0 Thanks for creating the instrumentation test cases, the code and test are good, can you please improve the commit history? there are some unrelevent commits.

**Review by MohitMaliFtechiz — APPROVED** [review]
@shashank-0-0 Thanks, LGTM

**Comment by kelson42** [comment]
CI does not pass

**Comment by MohitMaliFtechiz** [comment]
@shashank-0-0 Can you please rebase? @kelson42 The CI is failing due to  https://github.com/kiwix/kiwix-android/issues/4073 which we already fixed.

**Pull request merged**